File: VB/WinWebSolution.Module/ShowDetailViewFromListViewController.vb
Imports System.ComponentModel
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.DC
Imports DevExpress.ExpressApp.Model
Imports DevExpress.ExpressApp.SystemModule

Namespace WinWebSolution.Module
    Public Class ShowDetailViewFromListViewController
        Inherits ViewController(Of ListView)
        Implements IModelExtender

        Public Const EnabledKeyShowDetailView As String = "ShowDetailViewFromListViewController"
        Protected Overrides Sub OnActivated()
            MyBase.OnActivated()
            Dim targetController As ListViewProcessCurrentObjectController = Frame.GetController(Of ListViewProcessCurrentObjectController)()
            If targetController IsNot Nothing Then
                targetController.ProcessCurrentObjectAction.Enabled(EnabledKeyShowDetailView) = DirectCast(View.Model, IModelShowDetailView).ShowDetailView
            End If
        End Sub
        Public Sub ExtendModelInterfaces(ByVal extenders As DevExpress.ExpressApp.Model.ModelInterfaceExtenders) Implements IModelExtender.ExtendModelInterfaces
            extenders.Add(Of IModelViews, IModelDefaultShowDetailView)()
            extenders.Add(Of IModelListView, IModelShowDetailView)()
        End Sub
    End Class
    Public Interface IModelDefaultShowDetailView
        Inherits IModelNode

        <DefaultValue(True)> _
        Property DefaultShowDetailViewFromListView() As Boolean
    End Interface
    Public Interface IModelShowDetailView
        Inherits IModelNode

        Property ShowDetailView() As Boolean
    End Interface
    <DomainLogic(GetType(IModelShowDetailView))> _
    Public Class ModelShowDetailViewLogic
        Public Shared Function Get_ShowDetailView(ByVal showDetailView As IModelShowDetailView) As Boolean
            Dim defaultShowDetailViewFromListView As IModelDefaultShowDetailView = TryCast(showDetailView.Parent, IModelDefaultShowDetailView)
            If defaultShowDetailViewFromListView IsNot Nothing Then
                Return defaultShowDetailViewFromListView.DefaultShowDetailViewFromListView
            End If
            Return True
        End Function
    End Class
End Namespace

File: CS/WinWebSolution.Module/ShowDetailViewFromListViewController.cs
using System.ComponentModel;
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.DC;
using DevExpress.ExpressApp.Model;
using DevExpress.ExpressApp.SystemModule;

namespace WinWebSolution.Module {
    public class ShowDetailViewFromListViewController : ViewController<ListView>, IModelExtender {
        public const string EnabledKeyShowDetailView = "ShowDetailViewFromListViewController";
        protected override void OnActivated() {
            base.OnActivated();
            ListViewProcessCurrentObjectController targetController = Frame.GetController<ListViewProcessCurrentObjectController>();
            if (targetController != null) {
                targetController.ProcessCurrentObjectAction.Enabled[EnabledKeyShowDetailView] = ((IModelShowDetailView)View.Model).ShowDetailView;
            }
        }
        public void ExtendModelInterfaces(DevExpress.ExpressApp.Model.ModelInterfaceExtenders extenders) {
            extenders.Add<IModelViews, IModelDefaultShowDetailView>();
            extenders.Add<IModelListView, IModelShowDetailView>();
        }
    }
    public interface IModelDefaultShowDetailView : IModelNode {
        [DefaultValue(true)]
        bool DefaultShowDetailViewFromListView { get; set; }
    }
    public interface IModelShowDetailView : IModelNode {
        bool ShowDetailView { get; set; }
    }
    [DomainLogic(typeof(IModelShowDetailView))]
    public static class ModelShowDetailViewLogic {
        public static bool Get_ShowDetailView(IModelShowDetailView showDetailView) {
            IModelDefaultShowDetailView defaultShowDetailViewFromListView = showDetailView.Parent as IModelDefaultShowDetailView;
            if (defaultShowDetailViewFromListView != null) {
                return defaultShowDetailViewFromListView.DefaultShowDetailViewFromListView;
            }
            return true;
        }
    }
}